<launch>

    <!-- Launch recorder node -->
    <node pkg="rosbag_recorder" type="rosbag_recorder" name="rosbag_recorder_node" output="screen" respawn="true">
        <!-- Directory containing configuration files for the bag launcher to read -->
        <param name="configuration_directory"   type="string"       value="$(find rosbag_recorder)/config"/>
        <!-- Directory for the bags to be recorded to -->
        <param name="data_directory"            type="string"       value="/media/data/"/>
        <!-- Topic for bag launcher to listen to for launching bags -->
        <param name="start_bag_topic"           type="string"       value="/record/start"/>
        <!-- Topic for bag launcher to listen to for stopping bags -->
        <param name="stop_bag_topic"            type="string"       value="/record/stop"/>
        <!-- Boolean whether or not the bag name is published to a topic -->
        <param name="publish_name"              type="bool"         value="true"/>
        <!-- Topic to publish bag name to if publish_name is true -->
        <param name="name_topic"                type="string"       value="/record/bag_name"/>
        <!-- Boolean whether or not a heartbeat is published for each bag -->
        <param name="publish_heartbeat"         type="bool"         value="true"/>
        <!-- Topic to publish heartbeat to -->
        <param name="heartbeat_topic"           type="string"       value="/record/heartbeat"/>
        <!-- Interval in seconds on which to publish bag heartbeats -->
        <param name="heartbeat_interval"        type="double"       value="10.0"/>
    </node>

</launch>
